modules/standard/Reflection.chpl

```diff
@@ -75,7 +73,32 @@ proc getField(x:?t, param s:string) {
   return __primitive("field value by num", x, i);
 }
 
-// TODO ref return version of getField(x, string)
+/* Get a mutable ref to the ith field in a class or record.
+   Causes a compilation error if `i` is not in 1..numFields(t)
+   or if the argument is not mutable.
+
+   :arg x: a class or record
+   :arg i: which field to get
+   :returns: an rvalue referring to that field.
+ */
+inline
+proc getFieldRef(ref x:?t, param i:int) ref
+  return __primitive("field value by num", x, i);
+
+/* Get a mutable ref to a field in a class or record by name.
+   Will generate a compilation error if a field with that name
+   is not found or if the class or record is not mutable.
+
+   :arg x: a class or record
+   :arg s: the name of a field
+   :returns: an rvalue referring to that field.
+ */
+proc getFieldRef(ref x:?t, param s:string) ref {
+  param i = __primitive("field name to num", t, s);
+  if i == 0 then
+    compilerError("field ", s, " not found in ", t:string);
+  return __primitive("field value by num", x, i);
+}
```

Comment:
Ideally, it'd be nice to have a test (or param option + compopts for the existing tests) to try statements that would check these error conditions (field num = 0, < 0, > numFields;  field name not in record) in order to guarantee the advertised behavior and lock in the error messages.